exporters/committime/collector_bitbucket.py

```diff
@@ -44,54 +44,49 @@ class Version1(APIVersion):
     def test_url(self, server: str) -> str:
         return pelorus.url_joiner(server, self.root, self.test_path)
 
-    def commit_url(self, metric: CommitMetric) -> str:
+    def commit_url(self, info: CommitInfo) -> str:
         "Handle the URL for v1 specially."
 
-        git_server = metric.git_server
-        sha = metric.commit_hash
+        original_repo, sha = info
 
-        # URL munging copied from original code.
-        # TODO: this is messy. We should investigate the parsing that CommitMetric is doing.
+        git_server = original_repo.server
+
+        # URL munging adapted from older code.
 
         # Due to the BB V1 git pattern differences, need remove '/scm' and parse again.
-        old_url = metric.repo_url
+        old_url = original_repo.url
         # Parse out the V1 /scm, for whatever reason why it is present.
         new_url = old_url.replace("/scm", "")
-        # set the new url, so the parsing will happen
-        metric.repo_url = new_url
-        # set the new project name
-        project_name = metric.repo_project
-        # set the new group
-        group = metric.repo_group
-        # set the URL back to the original
-        metric.repo_url = old_url
+        repo = GitRepo.from_url(new_url)
 
         return pelorus.url_joiner(
             git_server,
             self.root,
-            self.pattern.format(group=group, project=project_name, commit=sha),
+            self.pattern.format(group=repo.group, project=repo.project, commit=sha),
         )
 
-    def update_metric_from_api(self, metric: CommitMetric, api_response: dict):
+    def timestamp_from_api(self, info: CommitInfo, api_response: dict) -> datetime:
+        ...
         # API V1 uses unix time
         commit_timestamp = api_response["committerTimestamp"]
 
         # Convert timestamp from miliseconds to seconds
         converted_timestamp = commit_timestamp / 1000
 
         timestamp = datetime.fromtimestamp(converted_timestamp, tz=timezone.utc)
+        return timestamp
 
-        logging.debug(
-            "API v1 returned sha: %s, timestamp: %s (%s)",
-            metric.commit_hash,
-            timestamp,
-            converted_timestamp,
-        )
-        # set the timestamp in the metric
-        metric.commit_timestamp = converted_timestamp
+        # logging.debug(
+        #     "API v1 returned sha: %s, timestamp: %s (%s)",
+        #     metric.commit_hash,
+        #     timestamp,
+        #     converted_timestamp,
+        # )
+        # # set the timestamp in the metric
+        # metric.commit_timestamp = converted_timestamp
 
-        # convert the time stamp to datetime and set in metric
-        metric.commit_time = timestamp.isoformat()
+        # # convert the time stamp to datetime and set in metric
+        # metric.commit_time = timestamp.isoformat()
```

Comment:
I suppose I didn't address the commented-out logs.

I think we should consider where and when debug logs are necessary, and how much context we need. For instance, does each line need to repeat what build or repo or namespace something is in, if we just have log lines like "begin inspecting $namespace" and "end inspecting $namespace"? 

I want to eventually consider [contextual, structured logging](https://github.com/konveyor/pelorus/issues/420) for this reason.